tests/features/camel-azure-storage-blob/src/main/java/org/apache/karaf/camel/test/CamelAzureStorageBlobRouteSupplier.java

```diff
@@ -0,0 +1,96 @@
+import org.apache.camel.component.azure.storage.blob.BlobConfiguration;
+import org.apache.camel.component.azure.storage.blob.BlobOperationsDefinition;
+import org.apache.camel.model.RouteDefinition;
+import org.apache.karaf.camel.itests.AbstractCamelSingleFeatureResultMockBasedRouteSupplier;
+import org.apache.karaf.camel.itests.CamelRouteSupplier;
+import org.osgi.service.component.annotations.Component;
+
+import com.azure.core.http.policy.HttpLogDetailLevel;
+import com.azure.core.http.policy.HttpLogOptions;
+import com.azure.storage.blob.BlobServiceClient;
+import com.azure.storage.blob.BlobServiceClientBuilder;
+import com.azure.storage.common.StorageSharedKeyCredential;
+
+@Component(name = "karaf-camel-azure-storage-blob-test", immediate = true, service = CamelRouteSupplier.class)
+public class CamelAzureStorageBlobRouteSupplier extends AbstractCamelSingleFeatureResultMockBasedRouteSupplier {
+
+    private static final String TEST_BLOB_CONTAINER_NAME = "test-container";
+    private static final String TEST_BLOB_NAME = "hello.txt";
+    private static final String TEST_ACCOUNT = System.getProperty("azure.accountName");
+
+    @Override
+    public void configure(CamelContext camelContext) {
+
+        final String host = System.getProperty("azure.host");
+        final String port = System.getProperty("azure.port");
+        final StorageSharedKeyCredential credential =
+                new StorageSharedKeyCredential(TEST_ACCOUNT, System.getProperty("azure.accountKey"));
+        final String endpoint = "http://" + host + ":" + port + "/" + TEST_ACCOUNT;
+
+        final BlobServiceClient blobServiceClient = new BlobServiceClientBuilder().endpoint(endpoint)
+                .credential(credential)
+                .httpLogOptions(new HttpLogOptions().setLogLevel(HttpLogDetailLevel.BODY_AND_HEADERS).setPrettyPrintBody(true))
+                .buildClient();
+
+        final BlobConfiguration configuration = new BlobConfiguration();
+        configuration.setServiceClient(blobServiceClient);
+        configuration.setAccountName(TEST_ACCOUNT);
+        configuration.setCredentials(credential);
+        configuration.setCredentialType(SHARED_KEY_CREDENTIAL);
+
+        final BlobComponent blobComponent = new BlobComponent();
+        blobComponent.setConfiguration(configuration);
+
+        camelContext.addComponent("azure-storage-blob", blobComponent);
+    }
+
+    @Override
+    protected Function<RouteBuilder, RouteDefinition> consumerRoute() {
+        return builder -> builder.fromF("azure-storage-blob://%s/%s?blobName=%s", TEST_ACCOUNT,
+                        TEST_BLOB_CONTAINER_NAME, TEST_BLOB_NAME)
+                .setHeader(BLOB_CONTAINER_NAME, builder.constant(TEST_BLOB_CONTAINER_NAME))
+                .setHeader(BLOB_NAME, builder.constant(TEST_BLOB_NAME))
+                .log("Downloaded the blob with content: ${body}")
+                .setBody(builder.constant("OK"));
```

Comment:
Why do you set it to OK? Maybe you could test directly the content of the blob, WDYT?